datadayessentials/model_inference/tests/test_model_inference.py

```diff
@@ -0,0 +1,62 @@
+#add imports
+from ..model_inference import InferenceModel, lightgbmInferenceModel
+import unittest
+from unittest.mock import patch, MagicMock
+import pytest
+
+
+class TestInferenceModel(unittest.TestCase):
+    def setUp(self):
+        self.model = MagicMock()
+        self.model.model = MagicMock()
+        self.model.feature_names_ = ["a", "b", "c"]
+        self.model.feature_importances_ = [0.1, 0.2, 0.3]
+        self.inference_model = InferenceModel(self.model)
+
+    def test_init(self):
+        self.assertEqual(self.inference_model.model, self.model.model)
+        self.assertEqual(
+            self.inference_model.feature_names_, self.model.feature_names_
+        )
+        self.assertEqual(
+            self.inference_model.feature_importances_,
+            self.model.feature_importances_,
+        )
+
+    def test_predict(self):
+        self.inference_model.model.predict = MagicMock(return_value=[1, 2, 3])
+        X = MagicMock()
+        self.assertEqual(self.inference_model.predict(X), [1, 2, 3])
+
+    def test_predict_proba(self):
+        self.inference_model.model.predict_proba = MagicMock(return_value=[1, 2, 3])
+        X = MagicMock()
+        self.assertEqual(self.inference_model.predict_proba(X), [1, 2, 3])
+
+class TestlightgbmInferenceModel(unittest.TestCase):
+    def setUp(self):
+        self.model = MagicMock()
+        self.model.model = MagicMock()
+        self.model.feature_names_ = ["a", "b", "c"]
+        self.model.feature_importances_ = [0.1, 0.2, 0.3]
+        self.inference_model = lightgbmInferenceModel(self.model)
```

Comment:
Can we create a lgbm model here to make sure the apis work?

Or Maybe use the LightGBM model class to set the spec for the MagicMock